optuna/visualization/_optimization_history.py

```diff
@@ -123,79 +218,48 @@ def _get_optimization_histories_with_error_bar(
     target_name: str,
     layout: "go.Layout",
 ) -> "go.Figure":
-    max_trial_number = np.max(
-        [
-            trial.number
-            for study in studies
-            for trial in study.get_trials(states=(TrialState.COMPLETE,))
-        ]
-    )
 
-    _target: Callable[[FrozenTrial], float]
-    if target is None:
-
-        def _target(t: FrozenTrial) -> float:
-            return cast(float, t.value)
-
-    else:
-        _target = target
+    eb_info = _get_optimization_history_error_bar_info(studies, target, target_name)
```

Comment:
For testability, maybe this function could receive `_OptimizationHistoryErrorBarInfo` directly. This way, we could test `_get_optimization_history_error_bar_info` and `_get_optimization_histories_with_error_bar` separately, i.e., we do not need to make `studies` to test `_get_optimization_histories_with_error_bar`.